google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/instance/AppEngineServerModel.java

```diff
@@ -124,107 +138,506 @@ public int getDefaultPort() {
+  @Override
+  public String getHost() {
+    return null;
   }
 
-  public void setArtifact(@Nullable Artifact artifact) {
-    if (artifact != null) {
-      myArtifactPointer = ArtifactPointerManager.getInstance(myCommonModel.getProject())
-          .createPointer(artifact);
-    } else {
-      myArtifactPointer = null;
-    }
+  public void setHost(String host) {
+    settings.setHost(host);
   }
 
-  public static class AppEngineModelSettings {
+  @Override
+  public Integer getPort() {
+    return settings.getPort();
+  }
+
+  public void setPort(Integer port) {
+    settings.setPort(port);
+  }
+
+  @Override
+  public String getAdminHost() {
+    return settings.getAdminHost();
+  }
+
+  public void setAdminHost(String adminHost) {
+    settings.setAdminHost(adminHost);
+  }
+
+  @Override
+  public Integer getAdminPort() {
+    return settings.getAdminPort();
+  }
+
+  public void setAdminPort(Integer adminPort) {
+    settings.setAdminPort(adminPort);
+  }
+
+  @Override
+  public String getAuthDomain() {
+    return settings.getAuthDomain();
+  }
+
+  public void setAuthDomain(String authDomain) {
+    settings.setAuthDomain(authDomain);
+  }
+
+  @Override
+  public String getStoragePath() {
+    return settings.getStoragePath();
+  }
+
+  public void setStoragePath(String storagePath) {
+    settings.setStoragePath(storagePath);
+  }
+
+  @Override
+  public String getLogLevel() {
+    return settings.getLogLevel();
+  }
+
+  public void setLogLevel(String logLevel) {
+    settings.setLogLevel(logLevel);
+  }
+
+  @Override
+  public Integer getMaxModuleInstances() {
+    return settings.getMaxModuleInstances();
+  }
+
+  public void setMaxModuleInstances(Integer maxModuleInstances) {
+    settings.setMaxModuleInstances(maxModuleInstances);
+  }
+
+  @Override
+  public Boolean getUseMtimeFileWatcher() {
+    return settings.isUseMtimeFileWatcher();
+  }
+
+  public void setUseMtimeFileWatcher(Boolean useMtimeFileWatcher) {
+    settings.setUseMtimeFileWatcher(useMtimeFileWatcher);
+  }
+
+  @Override
+  public String getThreadsafeOverride() {
+    return settings.getThreadsafeOverride();
+  }
+
+  public void setThreadsafeOverride(String threadsafeOverride) {
+    settings.setThreadsafeOverride(threadsafeOverride);
+  }
+
+  @Override
+  public String getPythonStartupScript() {
+    return settings.getPythonStartupScript();
+  }
+
+  public void setPythonStartupScript(String pythonStartupScript) {
+    settings.setPythonStartupScript(pythonStartupScript);
+  }
+
+  @Override
+  public String getPythonStartupArgs() {
+    return settings.getPythonStartupArgs();
+  }
+
+  public void setPythonStartupArgs(String pythonStartupArgs) {
+    settings.setPythonStartupArgs(pythonStartupArgs);
+  }
+
+  private static final String DELIMITER = "!@#";
+
+  @Override
+  public List<String> getJvmFlags() {
+    return Arrays.asList(settings.getJvmFlags() != null
+        ? settings.getJvmFlags().split(DELIMITER) : new String[]{});
```

Comment:
also random though: Do the other run configs split their flags? Since the user enters them as a free form string I wonder if it's not safest to just store exactly what they have inputted.